src/core/mount.c

```diff
@@ -1712,8 +1712,10 @@ static int mount_setup_unit(
                 return log_warning_errno(r, "Failed to set up mount unit for '%s': %m", where);
 
         /* If the mount changed properties or state, let's notify our clients */
-        if (flags & (MOUNT_PROC_JUST_CHANGED|MOUNT_PROC_JUST_MOUNTED))
+        if (flags & (MOUNT_PROC_JUST_CHANGED|MOUNT_PROC_JUST_MOUNTED)) {
+                device_found_node(m, what, DEVICE_FOUND_MOUNT, DEVICE_FOUND_MOUNT);
                 unit_add_to_dbus_queue(u);
```

Comment:
so, this swaps around how device units are instantiated. previously, the device node once it showed up in /proc/self/mountinfo or /proc/swaps was marked as existing ("tentative") via device_found_node(), and then we'd update the mount unit afterwards, to dep on it.

But with this change you swap this around. Which shouldn't be the big difference, but is still a bit weird I think